instrumentation/github.com/gin-gonic/gin/otelgin/option.go

```diff
@@ -86,8 +87,7 @@ func WithGinFilter(f ...GinFilter) Option {
 
 // WithSpanNameFormatter takes a function that will be called on every
 // request and the returned string will become the Span Name.
-func WithSpanNameFormatter(f func(r *http.Request) string) Option {
-	return optionFunc(func(c *config) {
-		c.SpanNameFormatter = f
-	})
+// Deprecated: since 0.58.0, remove in 0.59.0.
```

Comment:
It's helpful for deprecations to shortly explain their reason.

```suggestion
// Deprecated: [0.58.0] the span name now follows semantic conventions.
```

As for the version name, I've switched this to the deprecation strategy used by the collector.
Specify the first version where it's deprecated, since deprecations should always only last for one version.